packages/lisk-transactions/test/4_multisignature_transaction.ts

```diff
@@ -407,4 +409,119 @@ describe('Multisignature transaction class', () => {
+		});
+
+		it('should fail when valid signature already present and sent again', async () => {
+			const { status: arrangeStatus } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+
+			const { status, errors } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+			const expectedError = 'Encountered duplicate signature in transaction';
+
+			expect(arrangeStatus).to.eql(Status.PENDING);
+			expect(status).to.eql(Status.FAIL);
+			expect(errors[0].message).to.be.eql(expectedError);
+			expect(multisigTrs.signatures).to.include(membersSignatures[0].signature);
+		});
+
+		it('should fail to add invalid signature to transaction', async () => {
+			const { status, errors } = multisigTrs.addMultisignature(store, {
+				transactionId: multisigTrs.id,
+				publicKey:
+					'bb7ef62be03d5c195a132efe82796420abae04638cd3f6321532a5d33031b30c',
+				signature:
+					'eeee799c2d30d2be6e7b70aa29b57f9b1d6f2801d3fccf5c99623ffe45526104b1f0652c2cb586c7ae201d2557d8041b41b60154f079180bb9b85f8d06b3010c',
+			});
+
+			const expectedError =
+				'Failed to add signature eeee799c2d30d2be6e7b70aa29b57f9b1d6f2801d3fccf5c99623ffe45526104b1f0652c2cb586c7ae201d2557d8041b41b60154f079180bb9b85f8d06b3010c.';
+
+			expect(status).to.eql(Status.FAIL);
+			expect(errors[0].message).to.be.eql(expectedError);
+			expect(multisigTrs.signatures).to.be.empty;
+		});
+
+		it('should fail with valid signature not part of the group', async () => {
+			const nonMemberSignature: SignatureObject = {
+				transactionId: multisigTrs.id,
+				publicKey:
+					'cba7d88c54f3844bbab2c64b712e0ba3144921fe7a76c5f9df80b28ab702a35b',
+				signature:
+					'35d9bca853353906fbc44b86918b64bc0d21daf3ca16e230aa59352976624bc4ce69ac339f08b45c5e926d60cfa81276778e5858ff2bd2290e40d9da59cc5f0b',
+			};
+
+			const expectedError =
+				"Public Key 'cba7d88c54f3844bbab2c64b712e0ba3144921fe7a76c5f9df80b28ab702a35b' is not a member.";
+
+			const { status, errors } = multisigTrs.addMultisignature(
+				store,
+				nonMemberSignature,
+			);
+
+			expect(status).to.eql(Status.FAIL);
+			expect(errors[0].message).to.be.eql(expectedError);
+			expect(multisigTrs.signatures).to.be.empty;
+		});
+
+		it('status should remain pending when invalid signature sent', async () => {
+			const { status: arrangeStatus } = multisigTrs.addMultisignature(
+				store,
+				membersSignatures[0],
+			);
+
+			const nonMemberSignature: SignatureObject = {
+				transactionId: multisigTrs.id,
+				publicKey:
+					'cba7d88c54f3844bbab2c64b712e0ba3144921fe7a76c5f9df80b28ab702a35b',
+				signature:
+					'35d9bca853353906fbc44b86918b64bc0d21daf3ca16e230aa59352976624bc4ce69ac339f08b45c5e926d60cfa81276778e5858ff2bd2290e40d9da59cc5f0b',
+			};
+
+			multisigTrs.addMultisignature(store, nonMemberSignature);
+
+			expect(arrangeStatus).to.eql(Status.PENDING);
+			expect(multisigTrs['_multisignatureStatus']).to.eql(Status.PENDING);
```

Comment:
```suggestion
			expect((multisigTrs as any)._multisignatureStatus).to.eql(Status.PENDING);
```

Reply:
I originally has it casted as any but I thought we wanted to avoid the use of `any` as much as possible?

Reply:
only for `test` it's ok =)